src/units/drawer.cpp

```diff
@@ -462,6 +469,40 @@ void unit_drawer::redraw_unit(const unit& u) const
 		halo_man.set_location(ac.unit_halo_, halo_x, halo_y);
 	}
 
+	const std::vector<std::string> halos_abilities = u.halo_abilities();
+	bool has_abil_halo = !ac.abil_halos_.empty() && ac.abil_halos_.front() && ac.abil_halos_.front()->valid();
+	if(!has_abil_halo && !halos_abilities.empty()) {
+		for(const std::string& halo_ab : halos_abilities){
+			ac.abil_halos_.push_back(halo_man.add(
+				halo_x, halo_y,
+				halo_ab + u.TC_image_mods(),
+				map_location(-1, -1)
+			));
+		}
+	}
+	if(has_abil_halo && (ac.abil_halos_ref_ != halos_abilities || halos_abilities.empty())){
+		for(halo::handle& abil_halo : ac.abil_halos_){
+			halo_man.remove(abil_halo);
+		}
+		ac.abil_halos_.clear();
+		if(!halos_abilities.empty()){
+			for(const std::string& halo_ab : halos_abilities){
+				ac.abil_halos_.push_back(halo_man.add(
+					halo_x, halo_y,
+					halo_ab + u.TC_image_mods(),
+					map_location(-1, -1)
+				));
+			}
+		}
+	}
+	else if(has_abil_halo){
```

Comment:
This should be on the same line as the closing brace of the `if` statement.

Reply:
fixed in https://github.com/wesnoth/wesnoth/pull/7856/commits/c8e017f35afec42b61b220332b1ef804f509d4b1